saleor/graphql/account/bulk_mutations.py

```diff
@@ -39,8 +42,31 @@ class Meta:
         description = "Deletes staff users."
         model = models.User
         permissions = (AccountPermissions.MANAGE_STAFF,)
-        error_type_class = AccountError
-        error_type_field = "account_errors"
+        error_type_class = StaffError
+        error_type_field = "staff_errors"
+
+    @classmethod
+    def perform_mutation(cls, _root, info, ids, **data):
+        instances = cls.get_nodes_or_error(ids, "id", User)
+        errors = cls.clean_instances(info, instances)
+        count = len(instances)
+        if not errors and count:
+            clean_instance_ids = [instance.pk for instance in instances]
+            qs = models.User.objects.filter(pk__in=clean_instance_ids)
+            cls.bulk_action(queryset=qs, **data)
+        else:
+            count = 0
+        return count, errors
+
+    @classmethod
+    def clean_instances(cls, info, users):
+        errors = defaultdict(list)
+        cls.check_if_users_can_be_deleted(info, users, "ids", errors)
+        cls.check_if_requestor_can_manage_users(info, users, "ids", errors)
+        cls.check_if_removing_left_not_manageable_permissions(
+            info, users, "ids", errors
+        )
+        return ValidationError(errors) if errors else {}
```

Comment:
Function can return either instance of ValidationError or a dictionary so that doesn't seem like a good pattern...

Reply:
I can do it here, or move to `perform_mutation`. But still `perform_mutation`will return `ValidationError` or empty dict. But this is how our `BaseBulkMutations` works. So refactor should be done in `BaseBulkMutations` first.